BankClient/Bank.cs

```diff
@@ -1,166 +1,129 @@
+
+    /// <summary>
+    /// Function saves filterList and call function for print
+    /// </summary>
+    /// <param name="inputUser">Takes <see cref="UserInformation"/> object for filter</param>
+    /// <returns>groupOfUsers</returns>
+    public List<UserInformation> GetAllUsersInformation(UserInformation inputUser)
     {
-        Console.WriteLine("Example:Dmitry,Lapshin,0950095521");
-        Console.Write("Fiter: ");
-        var userfilter = Console.ReadLine().Split(',');
-        foreach (var param in userfilter)
-        {
-            param.Trim();
-        }
+        var filterList = Users.Where(listUsers =>
+            (listUsers.FirstName == inputUser.FirstName || string.IsNullOrEmpty(inputUser.FirstName)) &&
+            (listUsers.LastName == inputUser.LastName || string.IsNullOrEmpty(inputUser.LastName)) &&
+            (listUsers.PhoneNumber == inputUser.PhoneNumber || string.IsNullOrEmpty(inputUser.PhoneNumber))).ToList();
+        return filterList;
+    }
 
-        if (userfilter[0] == "" && userfilter.Length == 1)
-        {
-            ShowAllUsers();
-        }
-        var filterlist = Users.Where(x => x.FirstName == userfilter[0] || x.LastName == userfilter[0] || x.Phonenumber== userfilter[0] || x.ID.ToString() == userfilter[0]).ToList();
-        for (int i = 0; i < userfilter.Length; i++)
-        {
-            filterlist = filterlist.Where(x => x.FirstName == userfilter[i] || x.LastName == userfilter[i] || x.Phonenumber == userfilter[i] ||  x.ID.ToString() == userfilter[0]).ToList();
-        }
+    /// <summary>
+    /// Gets one user
+    /// </summary>
+    /// <param name="inputUser">Takes <see cref="UserInformation"/> object for filter</param>
+    /// <returns>Nothing</returns>
+    public UserInformation GetFirstUserInformation(UserInformation inputUser)
+    {
+        var firstMatchedUserInformation= Users.FirstOrDefault(listUsers =>
+            (listUsers.FirstName == inputUser.FirstName || string.IsNullOrEmpty(inputUser.FirstName)) &&
+            (listUsers.LastName == inputUser.LastName || string.IsNullOrEmpty(inputUser.LastName)) &&
+            (listUsers.PhoneNumber == inputUser.PhoneNumber || string.IsNullOrEmpty(inputUser.PhoneNumber)));
+            
+        return firstMatchedUserInformation;
 
-        return filterlist;
+    }
+    /// <summary>
+    /// Search user by ID
+    /// </summary>
+    /// <returns>Return user or null</returns>
+    /// 
+    public UserInformation GetUserById(Guid id)
+    {
+        var result = Users.FirstOrDefault(x => x.Id.ToString() == id.ToString());
+        return result;
     }
 
-    public void GetAllUsersInformation(List<UserInformation> filterlist)
+    /// <summary>
+    /// Change user information
+    /// </summary>
+    /// <param name="id">Takes <see cref="UserInformation"/> object which has all user information</param>
+    /// <returns>Nothing</returns>
+    public void ChangeUserInformation(Guid id)
```

Comment:
This method should accept `UserInformation` as argument. 

**Do not use UserInterface in `Bank.cs`!**

Reply:
I don't really understand how not to use UserInterface if I need to assign new data, which user should input, in the list

Reply:
Ok